[aggregator][py] Improve error handling in py aggregator bindings

olivielpeau opened the pull request:
### What does this PR do?

Improve error handling in py aggregator bindings by being
as defensive as possible in this part of the code.

### Motivation

Not handling errors from the C python API can lead to weird
behaviors from the python interpreter.

### Additional Notes

See commit description

olivielpeau committed:
[aggregator][py] Improve error handling in py submission bindings

We try to handle all easily recoverable/ignorable errors in the
bindings directly (usually by logging the issue and continuing the
submission). The other errors are bubbled up to the python interpreter
which will then raise an exception.

All python C API calls have their errors handled or a comment
explaining why we can abstain from checking errors.

olivielpeau committed:
[py] Add basic README

[skip ci]

olivielpeau committed:
[py/check] Log as much info as possible on instance loading error

Log the whole traceback if there's one. If there's none log the pvalue.
As a last resort try just logging the ptype.

masci reviewed (approved):
a couple of comments but LGTM anyways

hush-hush reviewed (commented):
New exception handler looks much better than what we use to have !

olivielpeau committed:
[collector][py] Address review comments

* Use error log level in `api.go`
* Add tests for code handling exceptions when AgentCheck objects
are instantiated: one case where there is a ptraceback, and one case
where there is only a pvalue. I don't know how to make the py code
raise an exception with only a ptype unfortunately.
